### PR TITLE
feat: remove asset remotely (AR-2034)

### PR DESCRIPTION
**Pull request by Garzas**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

When message contains asset, then after deleting it delete also asset externally

### Issues

- after deleting message asset was still stored externally

### Solutions

Before marking message as removed 
- fetch message locally
- check if contains asset
- if yes remove asset remotely

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/asset/AssetApiImpl.kt [hunk 1]
```diff
@@ -74,6 +77,11 @@ class AssetApiImpl internal constructor(
             }
         }
 
+    override suspend fun deleteAsset(assetId: AssetId, assetToken: String?): NetworkResponse<Unit> =
```
Comment (yamilmedina) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/asset/AssetApiImpl.kt:80]: Would be nice if we add a test for this case 🤾

**Review by yamilmedina — APPROVED** [review]
Delicate work as always @Garzas ! 🍪  🚀  👏

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/DeleteMessageUseCase.kt [hunk 2]
```diff
@@ -48,13 +54,31 @@ class DeleteMessageUseCase(
...
-        }.onFailure {
-            kaliumLogger.withFeatureId(MESSAGES).w("delete message failure: $it")
-            if (it is CoreFailure.Unknown) {
-                it.rootCause?.printStackTrace()
-            }
         }
+            .onSuccess {
+                messageRepository.getMessageById(conversationId, messageId)
+                    .onSuccess { message ->
+                        val assetToRemove = when (message.content) {
```
Comment (Garzas) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/DeleteMessageUseCase.kt:61]: I will make it more readable in next PR

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/asset/AssetApiImpl.kt (hunk 1 above)
Comment (Garzas) [reply network/src/commonMain/kotlin/com/wire/kalium/network/api/asset/AssetApiImpl.kt:80]: I will add it in next PR with local asset removal 💪

**Pull request merged**